test/pylib/suite/python.py

```diff
@@ -244,3 +247,36 @@ def write_junit_failure_report(self, xml_res: ET.Element) -> None:
         if self.server_log_filename is not None:
             system_err = ET.SubElement(xml_res, 'system-err')
             system_err.text = read_log(self.server_log_filename)
+
+
+# Use cache to execute this function once per pytest session.
+@cache
+def add_host_option(parser: Parser) -> None:
+    parser.addoption("--host", default="localhost",
+                     help="Scylla server host to connect to")
+
+
+# Use cache to execute this function once per pytest session.
+@cache
+def add_scylla_cql_connection_options(parser: Parser) -> None:
+    """Add pytest options for Scylla CQL connection."""
+
+    cql_options = parser.getgroup("Scylla CQL connection options")
+    cql_options.addoption("--port", default="9042",
+                          help="Scylla CQL port to connect to")
+    cql_options.addoption("--ssl", action="store_true",
+                          help="Connect to CQL via an encrypted TLSv1.2 connection")
+
+
+# Use cache to execute this function once per pytest session.
+@cache
+def add_s3_options(parser: Parser) -> None:
+    """Options for tests which use S3 server (i.e., cluster/object_store and cqlpy/test_tools.py)"""
+
+    s3_options = parser.getgroup("S3 server settings")
+    s3_options.addoption('--s3-server-address')
+    s3_options.addoption('--s3-server-port', type=int)
+    s3_options.addoption('--aws-access-key')
+    s3_options.addoption('--aws-secret-key')
+    s3_options.addoption('--aws-region')
+    s3_options.addoption('--s3-server-bucket')
```

Comment:
Wouldn't it be better to move it to the main conftest? Or cache will handle the case that when we will execute several directories with adding s3 options?

Reply:
Yes `@cache` will help to execute each of these functions once per pytest session.